source/common/upstream/strict_dns_cluster.cc

```diff
@@ -138,8 +138,10 @@ void StrictDnsClusterImpl::ResolveTarget::startResolve() {
 
         std::chrono::milliseconds final_refresh_rate = parent_.dns_refresh_rate_ms_;
 
-        if (parent_.respect_dns_ttl_ && ttl_refresh_rate != std::chrono::seconds(0)) {
+        if (parent_.respect_dns_ttl_ && ttl_refresh_rate != std::chrono::seconds(0) &&
```

Comment:
Do we also need this for logical DNS? Otherwise LGTM.

Reply:
No, we don't need this for logical DNS. If no records returns, dns refresh ttl will be set to default value (5000ms). The reason why strict dns fails is that the init value is set to `std::chrono::seconds::max`.